src/Main.js

```diff
@@ -53,22 +54,39 @@ class Main extends Component {
   // get list of playlists and playlist list length. also attach invert event
   componentWillMount() {
     window.addEventListener('keydown', (e) => this.handleInvert(e))
-
-    const { activePage, per } = this.state
-
     Promise.all([
       this.API.getBlockCount(),
-      this.API.getPaginatedChannelContents(activePage, per),
+      this.API.getChannelContents(),
     ])
-      .then(([length, playlists]) => {
+      .then(([length, playlistChannel]) => {
+        const resortedPlaylists = {
+          ...playlistChannel,
+          contents: playlistChannel.contents.reverse()
+        }
         this.setState({
           playlistListLength: length,
-          playlists,
-          maxItemsInCurrentPage: this.getMaxItemsInCurrentPage(length, this.state.per),
+          playlistChannel: resortedPlaylists,
+          searchList: playlistChannel,
         })
       })
   }
 
+  togglePlaylistOrder = (resortedPlaylists) => {
+    this.setState({ resortedPlaylists: resortedPlaylists.reverse() })
+  }
+
+  applySearch = (predicate) => {
+    const updatedList = this.state.playlistChannel.contents.filter(item => {
+      const text = decode(`${item.user.full_name} / ${item.title}`)
+      return text.toLowerCase().search(predicate) !== -1
+    })
+    this.setState({ searchList: { ...this.state.playlistChannel, contents: updatedList } })
```

Comment:
Looks like it just gets replaced here, makes semi-sense to me (set the searchList to the entire channel contents to start)

Reply:
ah gotcha. funny to be reminded why you did something